src/error.rs

```diff
@@ -8,7 +8,7 @@ pub enum PngError {
     DeflatedDataTooLong(usize),
     TimedOut,
     NotPNG,
-    APNGNotSupported,
+    APNGOutOfOrder,
```

Comment:
Hm, is this a BC break? `APNGNotSupported` should have been removed in v9 but it wasn’t.

Reply:
Yes, it technically is, as it's a publicly viewable enum.

[`cargo semver-checks`](https://github.com/obi1kenobi/cargo-semver-checks) is a really useful tool to answer this sort of questions:

![image](https://github.com/user-attachments/assets/b467f830-d2f7-4366-ba27-445f73824ea9)

Reply:
Nice, I've added APNGNotSupported back in and it's not showing that issue. However, it is showing a bunch of issues to do with the recently-added `#[must_use]` to a number of functions, should we worry about this?
I wonder if we should this tool to our workflow...

Reply:
Personally, I'd ignore the lints related to `#[must_use]`, as they are too pedantic to matter in practice: they only matter under non-default linter configurations that are a bit of a bad idea to blindly set up. Nevertheless, it's interesting that the tool finds out and let us know about these, just in case my opinion is wrong :slightly_smiling_face:

Adding this tool to our workflow would be a good idea. I didn't read too much on how it computes a baseline for the semver checks though, so making sure that it always picks something sane in a workflow (even if it runs from a PR, or from a release tag...) would be great.

Reply:
Great, that’s kinda what I figured. Are we good to go with this PR then?

Reply:
Before that, I want to give your seemingly good changes a more proper, in-depth look. Please allow for a bit of time until I get around doing that :smile:

Reply:
Ah sorry, I just assumed you already had. No urgency, I appreciate the scrutiny 🙂